reconcile/utils/terraform/config_provider.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+from typing import Any, Iterable, Mapping, Optional
+from reconcile.utils.external_resource_spec import ExternalResourceSpecInventory
+from reconcile.utils.external_resources import PROVIDER_AWS, get_external_resource_specs
+from reconcile.utils.ocm import OCMMap
+
+from reconcile.utils.terrascript_aws_client import TerrascriptClient as Terrascript
+
+
+class UnknownProvisionProviderError(Exception):
+    pass
+
+
+@dataclass
+class TerraformConfigProvider:
```

Comment:
Is the idea that `Terrascript` is really `TerrascriptAwsClient` and when there is a `TerraformCloudflareClient` that would also be handled in this class? If so, would those all be initialized by this class?

I'm definitely interested in how we get the right abstraction here that will allow us to write new clients that are easy to test and minimize shared state. I'll start to think about what that could look like with other providers/clients and try to work backwards to this.

Reply:
Yes, i think that is the idea. I'm working towards having terraform-resources being provider agnostic, and TerraformConfigProvider being provider aware.

I will rename TerrascriptClient to TerrascriptAwsClient. Already started that move in https://github.com/app-sre/qontract-reconcile/pull/2440.

Let me know what you think!